### ISSUE TITLE
[Improvement]: Improve worker message passing to reduce memory leak

### ISSUE THREAD

**Issue by HindujaB** [ballerina-platform/ballerina-lang#41845]
### Description

$subject

### Describe your problem(s)

The following code which passes 10KB json data between 3 workers increases the memory usage over time. 
The heap dump shows a large count of `WorkerDataChannel` objects. 

```ballerina
import ballerina/time;

// json payload = some 10KB json value
public function main() returns error? {
    time:Utc utc = time:utcAddSeconds(time:utcNow(), 1800); // 30 mins running
    while (utc >= time:utcNow()) {
        check testWorkerMessaging();
    }
}

isolated function testWorkerMessaging() returns error? {

    worker w1 returns error? {
        json res1 = clientFunc1(payload);
        res1 -> w2;
    }

    worker w2 returns error? {
        json res1 = check <- w1;
        json res2 = clientFunc2(res1);
        res2 -> w3;
    }

    worker w3 returns error? {
        json res2 = check <- w2;
        json res3 = clientFunc3(res2);
        res3 -> function;
    }

    json res3 = check <- w3;
    check wait w3;
    return ();
}

type Rec record {|
    int i;

|};

isolated function clientFunc1(json payload1) returns json {
    return payload1;
}

isolated function clientFunc2(json payload1) returns json {
    return payload1;
}

isolated function clientFunc3(json payload1) returns json {
    return payload1;
}
```

Memory graph : 
<img width="1174" alt="Screenshot 2023-12-07 at 16 49 09" src="https://github.com/ballerina-platform/ballerina-lang/assets/28644893/243be6bf-e149-41e2-a589-6f575dc0118f">

Heap dump :
<img width="1166" alt="Screenshot 2023-12-07 at 16 48 59" src="https://github.com/ballerina-platform/ballerina-lang/assets/28644893/ecd57b9d-b334-4fb0-b937-bdb5e9941079">

Heap dump file : 
https://drive.google.com/file/d/1dXK7C2tZiPxXkHsfhEYhXgRaG7y_Datw/view?usp=sharing

### Describe your solution(s)

Clean up the channels & channel details.

### Related area

-> Runtime

**Comment by HindujaB** [comment]
After removing the redundant `WorkerDataChannel` and `ChannelDetails` instances, the memory usage is reduced to the following.
Memory graph :
<img width="1132" alt="Screenshot 2024-01-29 at 11 34 12" src="https://github.com/ballerina-platform/ballerina-lang/assets/28644893/958c3b17-6e15-4302-8e4f-d5e2ba59d918">

Heap dump :
<img width="1301" alt="Screenshot 2024-01-29 at 11 34 22" src="https://github.com/ballerina-platform/ballerina-lang/assets/28644893/9426b1f8-a605-4f09-bf51-4a1f971e861b">

**Comment by HindujaB** [comment]
This is now available in 2201.9.0.
